sqlmesh/core/engine_adapter/athena.py

```diff
@@ -0,0 +1,441 @@
+
+        if properties:
+            return exp.Properties(expressions=properties)
+
+        return None
+
+    def _truncate_table(self, table_name: TableName) -> None:
+        table = exp.to_table(table_name)
+        # Athena doesnt support TRUNCATE TABLE. The closest thing is DELETE FROM <table> but it only works on Iceberg
+        self.execute(f"DELETE FROM {table.sql(dialect=self.dialect, identify=True)}")
+
+    def _table_type(
+        self, table_properties: t.Optional[t.Dict[str, exp.Expression]] = None
+    ) -> t.Union[t.Literal["hive"], t.Literal["iceberg"]]:
+        """
+        Use the user-specified table_properties to figure out of this is a Hive or an Iceberg table
+        """
+        # if table_type is not defined or is not set to "iceberg", this is a Hive table
+        if table_properties and (table_type := table_properties.get("table_type", None)):
+            if "iceberg" in table_type.sql(dialect=self.dialect).lower():
+                return "iceberg"
+        return "hive"
+
+    @lru_cache()
+    def _query_table_type(
+        self, table_name: TableName
+    ) -> t.Union[t.Literal["hive"], t.Literal["iceberg"]]:
+        """
+        Hit the DB to check if this is a Hive or an Iceberg table
+        """
+        table_name = exp.to_table(table_name)
+        # Note: SHOW TBLPROPERTIES gets parsed by SQLGlot as an exp.Command anyway so we just use a string here
+        # This also means we need to use dialect="hive" instead of dialect="athena" so that the identifiers get the correct quoting (backticks)
+        for row in self.fetchall(f"SHOW TBLPROPERTIES {table_name.sql(dialect='hive')}"):
+            # This query returns a single column with values like 'EXTERNAL\tTRUE'
+            row_lower = row[0].lower()
+            if "external" in row_lower and "true" in row_lower:
+                return "hive"
+        return "iceberg"
+
+    def _table_location(
+        self,
+        table_properties: t.Optional[t.Dict[str, exp.Expression]],
+        table: exp.Table,
+    ) -> t.Optional[exp.LocationProperty]:
+        base_uri: str
+
+        # If the user has manually specified a `s3_base_location`, use it
+        if table_properties and "s3_base_location" in table_properties:
+            s3_base_location_property = table_properties.pop(
+                "s3_base_location"
+            )  # pop because it's handled differently and we dont want it to end up in the TBLPROPERTIES clause
+            if isinstance(s3_base_location_property, exp.Expression):
+                base_uri = s3_base_location_property.name
+            else:
+                base_uri = s3_base_location_property
+
+        elif self.s3_warehouse_location:
+            # If the user has set `s3_warehouse_location` in the connection config, the base URI is <s3_warehouse_location>/<catalog>/<schema>/
+            catalog_name = table.catalog if hasattr(table, "catalog") else None
```

Comment:
Looking at signatures we always expect `table` to be of type `exp.Table`. How could `catalog` and `db` attributes be missing, provided they are property methods in that class?

Reply:
Good catch, this didnt get updated when I tightened up the signature